PhysicsTools/Heppy/python/analyzers/gen/LHEWeightAnalyzer.py

```diff
@@ -19,28 +19,49 @@ class LHEWeightAnalyzer( Analyzer ):
     def __init__(self, cfg_ana, cfg_comp, looperName ):
         super(LHEWeightAnalyzer,self).__init__(cfg_ana,cfg_comp,looperName)
 
+        self.LHEWeightsNames=[]
+        self.lumiCounter=-1
+        self.currentLumi=0
+
     def declareHandles(self):
         super(LHEWeightAnalyzer, self).declareHandles()
         self.mchandles['LHEweights'] = AutoHandle('externalLHEProducer',
                                                   'LHEEventProduct',
                                                   mayFail=True,
                                                   fallbackLabel='source',
                                                   lazy=False )
+        if self.cfg_ana.useLumiInfo:
+            self.mchandles['GenInfos'] = AutoHandle('generator',
+                                                    'GenEventInfoProduct',
+                                                    mayFail=True,
+                                                    fallbackLabel='source',
+                                                    lazy=False )
+            self.genLumiHandle = Handle("GenLumiInfoHeader")
 
     def beginLoop(self, setup):
         super(LHEWeightAnalyzer,self).beginLoop(setup)
+        
+        if self.cfg_ana.useLumiInfo:
+            lumis = Lumis(self.cfg_comp.files)
+            for lumi in lumis:
+                if lumi.getByLabel('generator',self.genLumiHandle):
+                    weightNames = self.genLumiHandle.product().weightNames()
+                    for wn in weightNames:
+                        self.LHEWeightsNames.append(wn)
+                    break #so stupid to do it that way, crash otherwise
```

Comment:
please put that explanation in the comments, so that if somebody has to modify that code in the future they will know ( `#so stupid to do it that way, crash otherwise` doesn't explain much )